### DISCUSSION TITLE
Gridify code to convert sqaure grid to hexagonal grid

### DISCUSSION THREAD

**Discussion by kavege** [mtex-toolbox/mtex#1551] [Ask Anything]
Hii everyone,

I've written this code to convert a square grid to an hexagonal one.
this is my code but it doesn't work : 

sqUnitCell = abs(round(ebsd.unitCell,4))
minUnit = min(sqUnitCell(sqUnitCell>0))
hexunitCell = minUnit * [-1 -1/sqrt(3);0 -2/sqrt(3); 1 -1/sqrt(3); 1 1/sqrt(3); 0  2/sqrt(3); -1 1/sqrt(3)];
ebsd = ebsd.gridify('unitCell',hexunitCell);

my ebsd.unitCell is  : 
     0.0155    0.0155
   -0.0155    0.0155
   -0.0155   -0.0155
    0.0155   -0.0155

Can you help me with this issue?

Best regards,
Roméo

**Comment by talkingbird19** [comment]
I am not sure, but maybe u can try this line:
hexunitCell = minUnit * [-1 -1/sqrt(3); -1 1/sqrt(3) ; 0 2/sqrt(3); 1 1/sqrt(3); 1 -1/sqrt(3);0 -2/sqrt(3)];
